src/ml/dimensionality_reduction/pca.nim

```diff
@@ -6,15 +6,16 @@ import
   ../../tensor/tensor, ../../linear_algebra/linear_algebra
 
 
-proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2): Tensor[T] =
+proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2, transform: bool = true): tuple[results: Tensor[T], components: Tensor[T]] =
```

Comment:
Seems like I didn't add it as well, I would be grateful if you added the {.noInit.} pragma.

This has huge performance implication in Arraymancer (and for any library that returns a garbage collected type), see https://github.com/nim-lang/Nim/issues/7743. Otherwise Nim adds a useless GenericReset proc which is a complex proc to reset an ref object to prime state. Except that there is no need to reset a result value that will be reassigned/constructed in the proc.

```Nim
proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2, transform: bool = true): tuple[results: Tensor[T], components: Tensor[T]] {.noInit.}=
```

Reply:
Got it. This also reminded me I forgot to remove transform from the signature, which I am also fixing